### PR TITLE
cmds/core/basename: return with non zero status on error

### PR DESCRIPTION
**Pull request by binjip978**

### CODE REVIEW

File: cmds/core/basename/basename_test.go [hunk 1]
```diff
@@ -39,15 +41,19 @@ func TestRun(t *testing.T) {
...
 	// Table-driven testing
 	for _, tt := range tests {
 		t.Run(fmt.Sprintf("Using flags %s", tt.flags), func(t *testing.T) {
 			var out bytes.Buffer
-			run(&out, tt.flags)
+			err := run(&out, tt.flags)
+
+			if !errors.Is(err, tt.err) {
+				t.Fatalf("expected %v got %v", tt.err, err)
+			}
```
Comment (leongross) [inline cmds/core/basename/basename_test.go:56]:
```suggestion
			if err := run(&out, tt.flags); err != nil && !errors.Is(err, tt.err) {
			}
				t.Fatalf("expected %v got %v", tt.err, err)
			}
```

**Review by leongross — APPROVED** [review]
I prefer shorter error processing, but apart from that lgtm.

File: cmds/core/basename/basename.go [hunk 2]
```diff
@@ -10,37 +10,41 @@
...
-		usage(w)
+		return errUsage
 	}
 }
 
 func main() {
-	run(os.Stdout, os.Args[1:])
+	if err := run(os.Stdout, os.Args[1:]); err != nil {
+		if errors.Is(err, errUsage) {
+			fmt.Printf("%v\n", errUsage)
```
Comment (rminnich) [inline cmds/core/basename/basename.go:45]: could we use log.Fatal here?

File: cmds/core/basename/basename_test.go [hunk 3]
```diff
@@ -39,15 +41,19 @@ func TestRun(t *testing.T) {
 		},
 		{
 			flags: []string{},
-			out:   "Usage: basename NAME [SUFFIX]",
+			err:   errUsage,
```
Comment (rminnich) [inline cmds/core/basename/basename_test.go:44]: nice!

File: cmds/core/basename/basename_test.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/basename/basename_test.go:56]: the common form I've been recommended to use over the years is the got, want form
got %v, want %v
also, you can use errors.Is with nil, i.e., I believe the err != nil check is not needed. If you use errors.Is, you can also easily add tests where err is nil and should not be.

File: cmds/core/basename/basename.go (hunk 2 above)
Comment (binjip978) [reply cmds/core/basename/basename.go:45]: sure

File: cmds/core/basename/basename_test.go (hunk 1 above)
Comment (binjip978) [reply cmds/core/basename/basename_test.go]: I changed to got/want.

**Review by rminnich — APPROVED** [review]
